pkg/operator/ceph/object/user/controller.go

```diff
@@ -271,11 +271,18 @@ func (r *ReconcileObjectStoreUser) reconcile(request reconcile.Request) (reconci
 }
 
 func (r *ReconcileObjectStoreUser) reconcileCephUser(cephObjectStoreUser *cephv1.CephObjectStoreUser) (reconcile.Result, error) {
-	err := r.createCephUser(cephObjectStoreUser)
-	if err != nil {
-		return reconcile.Result{}, errors.Wrapf(err, "failed to create object store user %q", cephObjectStoreUser.Name)
+	var err error
+	if cephObjectStoreUser.Status.Phase == k8sutil.Created {
```

Comment:
Every time the operator is restarted, it will also start with a Created event so the operator can ensure all the resources are up-to-date. So we can't really change the behavior. Instead, the preferred pattern is to detect if it is already configured when querying Ceph. We already have that on line 293 where it checks for `object.ErrorCodeFileExists`. In that case, you could just call `UpdateCephUser()` instead of `GetUser()`

Reply:
@travisn is there any other reliable way to know this is updation than new creation than assuming existing of user. Because for ex if there is already a user was created with same name, another person is trying to create same user with different display name. For the display name this may matter much. But consider for [bucketpolicy](https://github.com/rook/rook/pull/5800) might be important

Reply:
Doesn't the `object.ErrorCodeFileExists` return code from ceph tell you if it already exists? Or are you saying that the display name is the only unique identifier that we currently have for the user?

Reply:
> Doesn't the `object.ErrorCodeFileExists` return code from ceph tell you if it already exists? Or are you saying that the display name is the only unique identifier that we currently have for the user?

Sorry the use case which I mentioned looks like illogically.